apicapi/apic_manager.py

```diff
@@ -459,7 +459,10 @@ def get_function_profile(self, switch, module, port, transaction=None):
                     host = hostlinks[0]['host']
                     links = self.db.get_switch_and_port_for_host(host)
                     for link in links:
-                        switch2, module2, port2 = link
+                        try:
+                            switch2, module2, port2 = link
```

Comment:
Since you don't seem to use if2 and pod2, this could just be:
 switch2, module2, port2 = link[0:3]

Reply:
will do.